onchain/rollups/test/foundry/consensus/quorum/QuorumFactory.t.sol

```diff
@@ -0,0 +1,161 @@
+
+        emit QuorumCreated(quorumValidators, quorum);
+
+        checkFactoryLogs(quorumValidators, quorum);
+    }
+
+    function testNewQuorumDeterministic(
+        uint256 _numValidators,
+        bytes32 _salt
+    ) public {
+        _numValidators = bound(_numValidators, 2, 50);
+        vm.assume(_numValidators >= 2 && _numValidators <= 50);
+
+        address[] memory quorumValidators = generateValidators(_numValidators);
+        uint256[] memory shares = generateShares(quorumValidators);
+
+        IHistory history = new History(msg.sender);
+
+        address precalculatedAddress = factory.calculateQuorumAddress(
+            quorumValidators,
+            shares,
+            history,
+            _salt
+        );
+
+        vm.recordLogs();
+
+        Quorum quorum = factory.newQuorum(
+            quorumValidators,
+            shares,
+            history,
+            _salt
+        );
+
+        emit QuorumCreated(quorumValidators, quorum);
+
+        // Precalculated address must match actual address
+        assertEq(precalculatedAddress, address(quorum));
+        checkFactoryLogs(quorumValidators, quorum);
+    }
+
+    function testAlreadyDeployedNewQuorumDeterministic(
+        uint256 _numValidators,
+        bytes32 _salt
+    ) public {
+        vm.assume(_numValidators > 1);
+        vm.assume(_numValidators < 50);
+
+        address[] memory quorumValidators = generateValidators(_numValidators);
+        uint256[] memory shares = generateShares(quorumValidators);
+
+        IHistory history = new History(msg.sender);
+
+        factory.newQuorum(quorumValidators, shares, history, _salt);
+
+        //Deploy already deployed quorum
+        vm.expectRevert();
+        factory.newQuorum(quorumValidators, shares, history, _salt);
+    }
+
+    // HELPER FUNCTIONS
+    function generateValidators(
+        uint256 _numValidators
+    ) internal pure returns (address[] memory) {
+        address[] memory validators = new address[](_numValidators);
+        for (uint256 i = 0; i < _numValidators; i++) {
+            validators[i] = vm.addr(i + 1);
+        }
+        return validators;
+    }
+
+    function generateShares(
+        address[] memory validators
+    ) internal pure returns (uint256[] memory) {
+        //generate a random number of shares for each validator
+        uint256[] memory shares = new uint256[](validators.length);
+        for (uint256 i; i < shares.length; ++i) {
+            uint256 share = uint256(
+                keccak256(abi.encodePacked(i, validators[i]))
+            ) % 100;
+            shares[i] = (share > 0) ? share : ++share;
+        }
+        return shares;
+    }
+
+    function checkFactoryLogs(
+        address[] memory _quorumValidators,
+        Quorum quorum
+    ) internal {
+        Vm.Log[] memory entries = vm.getRecordedLogs();
+        uint256 numOfQuorumsCreated;
+
+        for (uint256 i = 0; i < entries.length; i++) {
```

Comment:
We don't need to assign `0` to `i` because this is already its default value upon declaration.

```suggestion
        for (uint256 i; i < entries.length; i++) {
```